File: app/src/main/java/com/paradoxo/amadeus/cognicao/Processadora.java
package com.paradoxo.amadeus.cognicao;

import android.app.Activity;

import com.paradoxo.amadeus.modelo.Entidade;
import com.paradoxo.amadeus.modelo.Sentenca;

import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;

import static com.paradoxo.amadeus.util.Preferencias.getPrefString;

public class Processadora {
    Activity activity;
    Acionadora acionadora;

    public static final String PREF_NOME_IA = "nomeIa";

    public Processadora(Activity activity) {
        this.activity = activity;
        acionadora = new Acionadora(activity);
        EventBus.getDefault().register(this);
    }


    public void processarEntrada(String entrada){
        entrada = entrada.toLowerCase();
        entrada = entrada.replace("?", "").replace("!", "");
        entrada = entrada.replace(getPrefString(PREF_NOME_IA, activity),"");
        entrada = entrada.trim();

        acionadora.isAcao(entrada);
    }

    @Subscribe
    public void lidarResultadoProcessamento(Entidade entidade){
        Sentenca sentenca = new Sentenca("Via Event");
        EventBus.getDefault().post(sentenca);
    }

    @Override
    protected void finalize() throws Throwable {
        super.finalize();
        EventBus.getDefault().unregister(this);
    }
}

File: app/src/main/java/com/paradoxo/amadeus/cognicao/Processadora.kt
package com.paradoxo.amadeus.cognicao

import android.app.Activity
import com.paradoxo.amadeus.modelo.Entidade
import com.paradoxo.amadeus.modelo.Sentenca
import com.paradoxo.amadeus.util.Preferencias.getPrefString
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe

class Processadora(private val activity: Activity) {

    private val acionadora = Acionadora(activity)

    init {
        EventBus.getDefault().register(this)
    }

    fun processarEntrada(entrada: String) {
        val nomeIa = getPrefString(PREF_NOME_IA, activity).trim()
        var entradaOriginal = entrada.trim()

        if (nomeIa.isNotEmpty()) {
            entradaOriginal = entradaOriginal.replace(
                Regex(Regex.escape(nomeIa), RegexOption.IGNORE_CASE),
                ""
            ).trim()
        }

        val normalizada = entradaOriginal
            .lowercase()
            .replace("?", "")
            .replace("!", "")
            .trim()

        acionadora.isAcao(entradaOriginal, normalizada)
    }

    @Subscribe
    fun lidarResultadoProcessamento(entidade: Entidade) {
        EventBus.getDefault().post(Sentenca("Via Event"))
    }

    fun destruir() {
        EventBus.getDefault().unregister(this)
    }

    companion object {
        const val PREF_NOME_IA = "nomeIA"
    }
}
